src/store/AltStore.js

```diff
@@ -89,19 +89,26 @@ class AltStore {
 
   listen(cb) {
     if (!fn.isFunction(cb)) throw new TypeError('listen expects a function')
-    const { dispose } = this.transmitter.subscribe(cb)
-    this.subscriptions.push({ cb, dispose })
-    return () => {
-      this.lifecycle('unlisten')
-      dispose()
+
+    // Already listening
+    const existing = this.subscriptions.find((s) => s.cb === cb)
```

Comment:
Isn't this a breaking change?

What do we expect to happen if we did something like this:
```
function listenHandler() { /* do something */ }

MyStore.listen(listenHandler);
MyStore.listen(listenHandler);
```

I would think we'd expect the callback to fire twice if we attached it manually twice.

This does match the behavior of `addEventListener` though- adding the same callback to the same event multiple times doesn't seem to fire it more than once.

Reply:
Oh yea it is a different behavior from the original!


Based on the original unlisten code, it is also not intentional and I don't think we use it like this anywhere in our codebase.


Original listen: add A once, add A 2nd, and you get AA
Original unlisten: remove A once and you remove BOTH AA -- the behavior is not symmetrical


For our use case (and for sanity's sake) it is better to make this work like add/removeEventListener